libraries/SrcWrapper/src/stm32/low_power.c

```diff
@@ -223,12 +223,22 @@ void LowPower_stop(serial_t *obj)
   HAL_PWREx_EnableFastWakeUp();
 #endif
 #ifdef __HAL_RCC_WAKEUPSTOP_CLK_CONFIG
-  /* Select HSI as system clock source after Wake Up from Stop mode */
+  /* Select MSI or HSI as system clock source after Wake Up from Stop mode */
+#ifdef RCC_STOP_WAKEUPCLOCK_MSI
+  __HAL_RCC_WAKEUPSTOP_CLK_CONFIG(RCC_STOP_WAKEUPCLOCK_MSI);
+#else
   __HAL_RCC_WAKEUPSTOP_CLK_CONFIG(RCC_STOP_WAKEUPCLOCK_HSI);
+#endif
 #endif
 
   /* Enter Stop mode */
-  HAL_PWR_EnterSTOPMode(PWR_LOWPOWERREGULATOR_ON, PWR_STOPENTRY_WFI);
+#if defined(STM32L4xx)
+  if (WakeUpUart == NULL) {
```

Comment:
Here I think we could check if it is the LPUART because it is managed in STOP2 mode.